plutus-contract/src/Language/Plutus/Contract/Test.hs

```diff
@@ -52,7 +56,19 @@ import qualified Wallet.Emulator                       as EM
 
 import           Language.Plutus.Contract.Trace        as X
 
-type TracePredicate a = InitialDistribution -> Predicate (ContractTraceResult a)
+newtype PredF f a = PredF { unPredF :: a -> f Bool }
+
+instance Contravariant (PredF f) where
```

Comment:
Is this not derivable? Presumably there's an instance for `->`

Reply:
There isn't an instance for `->` :(

Reply:
Oh of course, it has to be the last argument :roll_eyes:

There is `Contravariant.Op`, so you could do deriving via that.

Reply:
Ah yeah I was looking for somthing like that, thanks.